src/core/maps/cell_occupancy_estimator.h

```diff
@@ -47,6 +56,73 @@ struct Occupancy {
 struct Beam {
   double x_st, y_st; ///< Coordinates of a start of the beam.
   double x_end, y_end; ///< Coordinates where the beam reaches an obstacle.
+
+  //methods
+  bool touches_rect(const Rectangle& bnds) const {
```

Comment:
passes through means that the beginning end ending are divided by a border. 
But this function checks whether a beam is passing through a cell, but doesn't contain any point inside cell except of points on border